File: java/src/com/rohitsh/learnarray/ValidMountainArray.java
package com.rohitsh.learnarray;

public class ValidMountainArray {
    /*
    Given an array of integers arr, return true if and only if it is a valid mountain array.

    Recall that arr is a mountain array if and only if:

    arr.length >= 3
    There exists some i with 0 < i < arr.length - 1 such that:
    arr[0] < arr[1] < ... < arr[i - 1] < arr[i]
    arr[i] > arr[i + 1] > ... > arr[arr.length - 1]
     */

    public static boolean validMountainArray(int[] arr) {
        var n = arr.length;
        if(n >= 3){
            var i = 0;

            while (i < n - 1 && (arr[i + 1] > arr[i])) {
                i++;
            }
            if(i == 0 || i == arr.length - 1) {
                return false;
            }

            while (i < n - 1 && (arr[i + 1] < arr[i])) {
                i++;
            }


            return i == n - 1;
        } else {
            return false;
        }
    }

    public static void main(String[] args) {

        var arr = new int[] {0, 3, 2, 1};

        System.out.println(validMountainArray(arr));
    }
}

File: c-sharp/arrays/ValidMountainArray.cs
/*
    Given an array of integers arr, return true if and only if it is a valid mountain array.

    Recall that arr is a mountain array if and only if:

    arr.length >= 3
    There exists some i with 0 < i < arr.length - 1 such that:
    arr[0] < arr[1] < ... < arr[i - 1] < arr[i]
    arr[i] > arr[i + 1] > ... > arr[arr.length - 1]

*/

namespace c_sharp.arrays
{
    public class ValidMountainArrayProblem
    {
        public static bool ValidMountainArray(int[] arr)
        {
            if(arr.Length < 3) return false;

            int index = 0;

            int n = arr.Length;

            while (index < n - 1 &&  (arr[index] < arr[index + 1])  )
            {
                index++;
            }

            if (index == n - 1) return false;

            while(index < n - 1 &&  (arr[index] > arr[index + 1])  )
            {
                index++;
                
            }

            return index == n - 1;
        }
    }
}
